src/lib/workflow-validator.ts

```diff
@@ -15,29 +15,39 @@
  *
  */
 
-import { DefinedError, ValidateFunction } from 'ajv';
+import { ValidateFunction } from 'ajv';
 import { Specification } from './definitions';
 import { validators } from './validators';
 
 export class WorkflowValidator {
   /** The validation errors after running validate(), if any */
-  errors: DefinedError[] | never[] = [];
+  errors: ValidationError[] | never[] = [];
   /** The validate function */
   private validateFn: ValidateFunction<Specification.Workflow>;
+
   /**
    * Creates a new WorkflowValidator for the provided workflow
    * @param {Workflow} workflow The workflow to validate
    */
   constructor(private workflow: Specification.Workflow) {
     this.validateFn = validators.get('Workflow') as ValidateFunction<Specification.Workflow>;
   }
+
   /**
    * Validates the workflow, populates the errors if any
    * @returns {boolean} If the workflow is valid or not
    */
   validate(): boolean {
     const isValid = this.validateFn(this.workflow);
-    this.errors = this.validateFn.errors as DefinedError[];
+    if (this.validateFn.errors) {
+      this.errors = this.validateFn.errors.map(
+        (error) => new ValidationError(`invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`)
+      );
+    }
     return isValid;
   }
 }
+
+export class ValidationError {
```

Comment:
This class should be in its own file.

It might be useful to keep a reference to the `ajv` error. Maybe use the error as constructor parameter and build the message in the constructor's body instead so we don't loose the original info?

Reply:
do you mean to create an `error` class with the same properties than `DefinedError`?

Reply:
I was thinking about this:

```typescript
import { DefinedError } from 'ajv';
export class ValidationError {
  readonly message: string;
  constructor(readonly error: DefinedError) {
    this.message = `invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`;
  }
}
```

Reply:
done, 

we are coupling here our API with a external library

Reply:
Yes, we have two runtime dependencies, ajv and js-yaml, and, in that case, ajv "bleeds out" in the public api of our library. Do you see that as a problem?

Reply:
no problem, 

one "solution" is to create our own class with the same properties but for now I think that it is ok

Reply:
```typescript
import { DefinedError } from 'ajv';
export class ValidationError extends DefinedError {
  readonly message: string;
  constructor(error: DefinedError) {
    Object.assign(this, error);
    this.message = `invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`;
  }
}
```
maybe ?

Reply:
I was thinking more in something like: 

```
export class Error{
  instancePath: string;
  schemaPath: string;
  message: string;
}


export class ValidationError {
  readonly message: string;
  
  constructor(readonly error: Error) {
    this.message = `invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`;
  }
}

```

Reply:
Maybe replicate more properties in case we need them at a point:
```typescript
export interface Error{
  keyword: K
  instancePath: string
  schemaPath: string
  params: P
  // Added to validation errors of "propertyNames" keyword schema
  propertyName?: string
  // Excluded if option `messages` set to false.
  message?: string
  // These are added with the `verbose` option.
  schema?: S
  parentSchema?: any
  data?: unknown
}
```

Reply:
we don't need them yet!! and this is an easy change, I would prefer to do it once we are using those properties if you don't mind